src/main/java/seedu/address/model/event/EventName.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.event;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class EventName {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String eventName;
+
+    /**
+     * Constructs a {@code EventName}.
+     *
+     * @param name A valid EventName.
+     */
+    public EventName(String name) {
+        requireNonNull(name);
+        checkArgument(isValidName(name), MESSAGE_CONSTRAINTS);
+        eventName = name;
+    }
+
+    /**
+     * Returns true if a given string is a valid name.
+     */
+    public static boolean isValidName(String test) {
```

Comment:
Same suggestion as above for the naming.
```suggestion
    public static boolean isValidName(String testName) {
```